### PR TITLE
[Merged by Bors] - feat: port Topology.Algebra.GroupWithZero

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Algebra/GroupWithZero.lean [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+import Mathlib.Topology.Algebra.Monoid
+import Mathlib.Algebra.Group.Pi
+import Mathlib.Topology.Homeomorph
+
+/-!
+# Topological group with zero
+
+In this file we define `HasContinuousInv₀` to be a mixin typeclass a type with `Inv` and
+`Zero` (e.g., a `GroupWithZero`) such that `λ x, x⁻¹` is continuous at all nonzero points. Any
+normed (semi)field has this property. Currently the only example of `HasContinuousInv₀` in
```
Comment (dupuisf) [inline Mathlib/Topology/Algebra/GroupWithZero.lean:20]:
```suggestion
`Zero` (e.g., a `GroupWithZero`) such that `fun x ↦ x⁻¹` is continuous at all nonzero points. Any
normed (semi)field has this property. Currently the only example of `HasContinuousInv₀` in
```

File: Mathlib/Topology/Algebra/GroupWithZero.lean [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+@[continuity]
+theorem Continuous.div_const (hf : Continuous f) (y : G₀) : Continuous fun x => f x / y := by
+  simpa only [div_eq_mul_inv] using hf.mul continuous_const
+#align continuous.div_const Continuous.div_const
+
+end DivConst
+
+/-- A type with `0` and `Inv` such that `λ x, x⁻¹` is continuous at all nonzero points. Any
+normed (semi)field has this property. -/
```
Comment (dupuisf) [inline Mathlib/Topology/Algebra/GroupWithZero.lean:85]:
```suggestion
/-- A type with `0` and `Inv` such that `fun x ↦ x⁻¹` is continuous at all nonzero points. Any
normed (semi)field has this property. -/
```

File: Mathlib/Topology/Algebra/GroupWithZero.lean [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+
+export HasContinuousInv₀ (continuousAt_inv₀)
+
+section Inv₀
+
+variable [Zero G₀] [Inv G₀] [TopologicalSpace G₀] [HasContinuousInv₀ G₀] {l : Filter α} {f : α → G₀}
+  {s : Set α} {a : α}
+
+/-!
+### Continuity of `λ x, x⁻¹` at a non-zero point
```
Comment (dupuisf) [inline Mathlib/Topology/Algebra/GroupWithZero.lean:99]:
```suggestion
/-!
### Continuity of `fun x ↦ x⁻¹` at a non-zero point
```

File: Mathlib/Topology/Algebra/GroupWithZero.lean [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+  continuousAt_inv₀ hx
+#align tendsto_inv₀ tendsto_inv₀
+
+theorem continuousOn_inv₀ : ContinuousOn (Inv.inv : G₀ → G₀) ({0}ᶜ) := fun _x hx =>
+  (continuousAt_inv₀ hx).continuousWithinAt
+#align continuous_on_inv₀ continuousOn_inv₀
+
+/-- If a function converges to a nonzero value, its inverse converges to the inverse of this value.
+We use the name `tendsto.inv₀` as `tendsto.inv` is already used in multiplicative topological
+groups. -/
```
Comment (dupuisf) [inline Mathlib/Topology/Algebra/GroupWithZero.lean:116]:
```suggestion
We use the name `Tendsto.inv₀` as `Tendsto.inv` is already used in multiplicative topological
groups. -/
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**